Bottom: 2878080687f6c4f2da144689e7abcc5922575616
Top:    19faef4fc5ec7bc98a9cf9cf9c8b6802898f97c0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:54 +0000

Rename v->e and e->v to clairfy edge vs. vertex usage


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 86fd69d..3aec238 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1069,15 +1069,15 @@ theState (VNODE * e)
 static void
 print_labels (PLINE * a)
 {
-  VNODE *c = &a->head;
+  VNODE *e = &a->head;
 
   do
     {
       DEBUGP ("%#mD->%#mD labeled %s\n",
-              EDGE_BACKWARD_VERTEX (c)->point[0], EDGE_BACKWARD_VERTEX (c)->point[1],
-               EDGE_FORWARD_VERTEX (c)->point[0],  EDGE_FORWARD_VERTEX (c)->point[1], theState (c));
+              EDGE_BACKWARD_VERTEX (e)->point[0], EDGE_BACKWARD_VERTEX (e)->point[1],
+               EDGE_FORWARD_VERTEX (e)->point[0],  EDGE_FORWARD_VERTEX (e)->point[1], theState (e));
     }
-  while ((c = NEXT_EDGE (c)) != &a->head);
+  while ((e = NEXT_EDGE (e)) != &a->head);
 }
 #endif
 #endif
@@ -1092,17 +1092,17 @@ label_contour
 static BOOLp
 label_contour (PLINE * a)
 {
-  VNODE *cur = &a->head; /* cur is considered an edge */
+  VNODE *cure = &a->head; /* cur is considered an edge */
   VNODE *first_labelled = NULL;
   int label = UNKNWN;
 
   do
     {
-      if (cur->cvc_next)	/* examine cross vertex */
+      if (cure->cvc_next)	/* examine cross vertex */
 	{
-	  label = edge_label (cur);
+	  label = edge_label (cure);
 	  if (first_labelled == NULL)
-	    first_labelled = cur;
+	    first_labelled = cure;
 	  continue;
 	}
 
@@ -1111,9 +1111,9 @@ label_contour (PLINE * a)
 
       /* This labels nodes which aren't cross-connected */
       assert (label == INSIDE || label == OUTSIDE);
-      LABEL_EDGE (cur, label);
+      LABEL_EDGE (cure, label);
     }
-  while ((cur = NEXT_EDGE (cur)) != first_labelled);
+  while ((cure = NEXT_EDGE (cure)) != first_labelled);
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
@@ -1470,35 +1470,35 @@ typedef enum
 } DIRECTION;
 
 /* Jump Rule  */
-/* VNODE * is considered an edge */
-typedef int (*J_Rule) (char, VNODE *, DIRECTION *);
+/* e is considered an edge */
+typedef int (*J_Rule) (char p, VNODE *e, DIRECTION *cdir);
 
-/* v is considered an edge */
+/* e is considered an edge */
 static int
-IsectJ_Rule (char p, VNODE * v, DIRECTION * cdir)
+IsectJ_Rule (char p, VNODE *e, DIRECTION * cdir)
 {
   *cdir = FORW;
-  return (v->Flags.status == INSIDE || v->Flags.status == SHARED);
+  return (e->Flags.status == INSIDE || e->Flags.status == SHARED);
 }
 
-/* v is considered an edge */
+/* e is considered an edge */
 static int
-UniteJ_Rule (char p, VNODE * v, DIRECTION * cdir)
+UniteJ_Rule (char p, VNODE *e, DIRECTION * cdir)
 {
   *cdir = FORW;
-  return (v->Flags.status == OUTSIDE || v->Flags.status == SHARED);
+  return (e->Flags.status == OUTSIDE || e->Flags.status == SHARED);
 }
 
-/* v is considered an edge */
+/* e is considered an edge */
 static int
-XorJ_Rule (char p, VNODE * v, DIRECTION * cdir)
+XorJ_Rule (char p, VNODE *e, DIRECTION * cdir)
 {
-  if (v->Flags.status == INSIDE)
+  if (e->Flags.status == INSIDE)
     {
       *cdir = BACKW;
       return TRUE;
     }
-  if (v->Flags.status == OUTSIDE)
+  if (e->Flags.status == OUTSIDE)
     {
       *cdir = FORW;
       return TRUE;
@@ -1506,21 +1506,21 @@ XorJ_Rule (char p, VNODE * v, DIRECTION * cdir)
   return FALSE;
 }
 
-/* v is considered an edge */
+/* e is considered an edge */
 static int
-SubJ_Rule (char p, VNODE * v, DIRECTION * cdir)
+SubJ_Rule (char p, VNODE *e, DIRECTION * cdir)
 {
-  if (p == 'B' && v->Flags.status == INSIDE)
+  if (p == 'B' && e->Flags.status == INSIDE)
     {
       *cdir = BACKW;
       return TRUE;
     }
-  if (p == 'A' && v->Flags.status == OUTSIDE)
+  if (p == 'A' && e->Flags.status == OUTSIDE)
     {
       *cdir = FORW;
       return TRUE;
     }
-  if (v->Flags.status == SHARED2)
+  if (e->Flags.status == SHARED2)
     {
       if (p == 'A')
 	*cdir = FORW;
@@ -1537,29 +1537,29 @@ SubJ_Rule (char p, VNODE * v, DIRECTION * cdir)
  *
  * returns true if an edge is found, false otherwise
  */
-/* *cur is considered a vertex */
+/* *curv is considered a vertex */
 static int
-jump (VNODE ** cur, DIRECTION * cdir, J_Rule j_rule)
+jump (VNODE **curv, DIRECTION *cdir, J_Rule j_rule)
 {
   CVCList *d, *start;
   VNODE *e; /* e is considered an edge */
   DIRECTION newone;
 
-  if (!(*cur)->cvc_prev)	/* not a cross-vertex */
+  if (!(*curv)->cvc_prev)	/* not a cross-vertex */
     {
-      if ((*cdir == FORW) ? VERTEX_FORWARD_EDGE (*cur)->Flags.mark :
-                           VERTEX_BACKWARD_EDGE (*cur)->Flags.mark)
+      if ((*cdir == FORW) ? VERTEX_FORWARD_EDGE (*curv)->Flags.mark :
+                           VERTEX_BACKWARD_EDGE (*curv)->Flags.mark)
 	return FALSE;
       return TRUE;
     }
 #ifdef DEBUG_JUMP
-  DEBUGP ("jump entering node at %$mD\n", (*cur)->point[0], (*cur)->point[1]);
+  DEBUGP ("jump entering node at %$mD\n", (*curv)->point[0], (*curv)->point[1]);
 #endif
   /* Pick the descriptor of the edge we came into this vertex with, then spin (anti?)clock-wise one edge descriptor */
   if (*cdir == FORW)
-    d = (*cur)->cvc_prev->prev; /* If we start with a CVC Vertex.. this previous edge has not been vetted for whether it belongs in the polygon or not!! */
+    d = (*curv)->cvc_prev->prev; /* If we start with a CVC Vertex.. this previous edge has not been vetted for whether it belongs in the polygon or not!! */
   else
-    d = (*cur)->cvc_next->prev;
+    d = (*curv)->cvc_next->prev;
   start = d;
   do
     {
@@ -1582,7 +1582,7 @@ jump (VNODE ** cur, DIRECTION * cdir, J_Rule j_rule)
 		DEBUGP ("jump leaving node at %#mD\n",
 			EDGE_BACKWARD_VERTEX (e)->point[0], EDGE_BACKWARD_VERTEX (e)->point[1]);
 #endif
-	      *cur = d->parent;
+	      *curv = d->parent;
 	      *cdir = newone;
 	      return TRUE;
 	    }
@@ -1594,9 +1594,10 @@ jump (VNODE ** cur, DIRECTION * cdir, J_Rule j_rule)
 
 /* start is considered a vertex */
 static int
-Gather (VNODE * start, PLINE ** result, J_Rule j_rule, DIRECTION initdir)
+Gather (VNODE *startv, PLINE **result, J_Rule j_rule, DIRECTION initdir)
 {
-  VNODE *cur = start, *newn; /* cur is considered a vertex */
+  VNODE *curv = startv; /* curv is considered a vertex */
+  VNODE *newn;
   DIRECTION dir = initdir;
 #ifdef DEBUG_GATHER
   DEBUGP ("gather direction = %d\n", dir);
@@ -1605,10 +1606,10 @@ Gather (VNODE * start, PLINE ** result, J_Rule j_rule, DIRECTION initdir)
   do
     {
       /* see where to go next */
-      if (!jump (&cur, &dir, j_rule))
+      if (!jump (&curv, &dir, j_rule))
 	break;
-      /* add edge to polygon */
-      if ((newn = poly_CreateNode (cur->point)) == NULL)
+      /* add vertex to polygon */
+      if ((newn = poly_CreateNode (curv->point)) == NULL)
         return err_no_memory;
       if (!*result)
 	{
@@ -1621,31 +1622,31 @@ Gather (VNODE * start, PLINE ** result, J_Rule j_rule, DIRECTION initdir)
 	  poly_InclVertex (PREV_VERTEX (&(*result)->head), newn);
 	}
 #ifdef DEBUG_GATHER
-      DEBUGP ("gather vertex at %#mD\n", cur->point[0], cur->point[1]);
+      DEBUGP ("gather vertex at %#mD\n", curv->point[0], curv->point[1]);
 #endif
       /* Now mark the edge as included.  */
-      newn = (dir == FORW) ? VERTEX_FORWARD_EDGE (cur) : VERTEX_BACKWARD_EDGE (cur);
+      newn = (dir == FORW) ? VERTEX_FORWARD_EDGE (curv) : VERTEX_BACKWARD_EDGE (curv);
       newn->Flags.mark = 1;
       /* for SHARED edge mark both */
       if (newn->shared)
 	newn->shared->Flags.mark = 1;
 
       /* Advance to the next vertex.  */
-      cur = (dir == FORW) ? NEXT_VERTEX (cur) : PREV_VERTEX (cur);
+      curv = (dir == FORW) ? NEXT_VERTEX (curv) : PREV_VERTEX (curv);
     }
   while (1);
   return err_ok;
 }				/* Gather */
 
-/* cur is considered an edge */
+/* cure is considered an edge */
 static void
-Collect1 (jmp_buf * e, VNODE * cur, DIRECTION dir, POLYAREA ** contours,
-	  PLINE ** holes, J_Rule j_rule)
+Collect1 (jmp_buf *e, VNODE *cure, DIRECTION dir, POLYAREA **contours,
+          PLINE **holes, J_Rule j_rule)
 {
   PLINE *p = NULL;		/* start making contour */
   int errc = err_ok;
-  if ((errc = Gather ((dir == FORW) ? EDGE_BACKWARD_VERTEX (cur) :
-                                      EDGE_FORWARD_VERTEX (cur),
+  if ((errc = Gather ((dir == FORW) ? EDGE_BACKWARD_VERTEX (cure) :
+                                      EDGE_FORWARD_VERTEX (cure),
                       &p, j_rule, dir)) != err_ok)
     {
       if (p != NULL)
@@ -1677,10 +1678,10 @@ static void
 Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes,
          J_Rule j_rule)
 {
-  VNODE *cur; /* cur is considered an edge */
+  VNODE *cure; /* cure is considered an edge */
   DIRECTION dir = UNINITIALISED;
 
-  cur = (&a->head);
+  cure = (&a->head);
 //  cur = (&a->head)->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
   do
     {
@@ -1689,14 +1690,14 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
       // In particular, consider the case when we collect a 'B' polygon contour.
       // Could some of that countour may already have been collected, and there
       // still be a piece we are interested in after? (Can we reach it though??)
-      if (cur->Flags.mark != 0)
+      if (cure->Flags.mark != 0)
         break;
 #endif
 
-      if (j_rule (poly, cur, &dir) && cur->Flags.mark == 0)
-	Collect1 (e, cur, dir, contours, holes, j_rule);
+      if (j_rule (poly, cure, &dir) && cure->Flags.mark == 0)
+	Collect1 (e, cure, dir, contours, holes, j_rule);
     }
-  while ((cur = NEXT_EDGE (cur)) != &a->head);
+  while ((cure = NEXT_EDGE (cure)) != &a->head);
 }				/* Collect */
